addons/test_import_export/tests/test_import.py

```diff
@@ -992,14 +998,83 @@ def test_xlsx_datetime_values_assigned_to_wrong_field(self):
                 'datetime_format': options.get('datetime_format'),
             }
         )
-        self.assertTrue(response.get('messages'), "Expected error messages due to date/time values in non-date fields")
-        self.assertEqual(len(response['messages']), 2)
+        self.assertFalse(response.get('messages'))
+        self.assertTrue(all(isinstance(name, str) for name in response['name']))
 
-        self.assertEqual(response['messages'][0]['type'], 'error')
-        self.assertIn("Field 'somevalue' does not accept date/time values.", response['messages'][0]['message'])
+    @unittest.skipUnless(can_import('xlwt') and can_import('openpyxl'), "xlwt/openpyxl not available")
+    def test_xlsx_datetime_values_assigned_to_related_char_field(self):
+        """Test that importing datetime values to a related char field is converted"""
+        file_content = generate_xlsx(
+            {
+                'Child/Name': ['foo', datetime.datetime(2020, 1, 6, 8, 10), datetime.date(2024, 7, 1)],
+            }
+        )
+        file_type = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+        import_wizard = self.env["base_import.import"].create({
+            'res_model': "import.o2m",
+            'file': file_content,
+            'file_type': file_type,
+        })
+        result = import_wizard.parse_preview({'has_headers': True})
+        options = result.get('options', {})
+        response = import_wizard.execute_import(
+            ["value/name"],
+            ["Child/Name"],
+            {
+                'quoting': '"', 'separator': ',', 'has_headers': True,
+                'date_format': options.get('date_format'), 'datetime_format': options.get('datetime_format'),
```

Comment:
I think it can be a good a idea to test with a specific `date_format`/`datetime_format` to ensure it works whatever the format. What do you think ?